mono/mini/mini-arm.c

```diff
@@ -5959,19 +5959,19 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			bb->spill_slot_defs = g_slist_prepend_mempool (cfg->mempool, bb->spill_slot_defs, ins);
 			break;
 		case OP_GC_SAFE_POINT: {
-#if defined (USE_COOP_GC)
-			const char *polling_func = NULL;
-			guint8 *buf [1];
+			if (mono_threads_is_coop_enabled ()) {
```

Comment:
An assert or this condition is actually a bad idea, as it means, in the AOT case, we have to compile the assembly with coop enabled, if we want to run the assembly with coop enabled. And I don't think that's a dependency we want to introduce, as it's pretty sneaky...
Same remark as for https://github.com/mono/mono/pull/2169/files#diff-307195a8afdfeac6e3df25eef5aaf9d6R1436: wouldn't it be better to always insert safepoints (for platforms that support it of course), and if at runtime we don't have coop suspend enabled, then mono_polling_required will never be set to 1, thus never calling mono_threads_state_poll.